plugins/modules/podman_login.py

```diff
@@ -107,6 +113,18 @@ def login(module, executable, registry, authfile,
             command.append('--tls-verify')
         else:
             command.append('--tls-verify=False')
+    # Use a checksum to check if the auth JSON has changed
+    checksum = None
+    docker_authfile = os.path.expandvars('$HOME/.docker/config.json')
+    if os.path.exists(authfile):
+        content = open(authfile, 'rb').read()
+        if bytes(registry, 'UTF-8') in content:
+            checksum = hashlib.md5(content).hexdigest()
+    # podman falls back to ~/.docker/config.json if the default authfile doesn't exist
+    elif os.path.exists(docker_authfile):
+        content = open(docker_authfile, 'rb').read()
```

Comment:
Instead of duplication, you can discover a file before and then to calculate the checksum, something like:

```python
check_file = authfile if os.path.exists(authfile) else docker_authfile
if os.path.exists(check_file):
        content = open(authfile, 'rb').read()
        if bytes(registry, 'UTF-8') in content:
            checksum = hashlib.md5(content).hexdigest()

```

Reply:
Thanks for the suggestion, that's indeed much better (with a small change to open `check_file`)